g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/TranslationProductAction.java

```diff
@@ -93,7 +96,81 @@ public APIResponseDTO getMultTrans(String productName, String version,String pse
         translationDTO = multipleComponentsService.getMultiComponentsTranslation(translationDTO);
         return super.handleResponse(APIResponseStatus.OK, translationDTO);
     }
-
+    
+    
+    public APIResponseDTO getMultTrans(String productName, String version, String componentsStr, String localesStr, String pseudo,
+            HttpServletRequest req)  throws Exception {
+        if(StringUtils.isEmpty(componentsStr) && StringUtils.isEmpty(localesStr)) {
+            return  getAllCompTrans( productName,  version, pseudo, req) ;
+        }else {
+            return getV2MultipleComponentsTrans(productName, componentsStr,  version,  localesStr,  pseudo, req);
+        }
+        
+        
+    }
+   
+    /**
+     *  get the API v2 mult-component translation
+     */
+      public APIResponseDTO getV2MultipleComponentsTrans(String productName,
+               String components, String version, String locales, String pseudo,
+               HttpServletRequest req) throws Exception {
+           TranslationDTO translationDTO = new TranslationDTO();
+           translationDTO.setProductName(productName);
+           translationDTO.setVersion(version);
+           List<String> componentList = null;
+           if (StringUtils.isEmpty(components)) {
+               componentList = productService.getComponentNameList(productName, version);
+           }else {
+               componentList = new ArrayList<String>();
+               for (String component : components.split(",")) {
+                   componentList.add(component.trim());
+               }
+           }
+           translationDTO.setComponents(componentList);
+           List<String> localeList = new ArrayList<String>();
+           if (new Boolean(pseudo)) {
+               localeList.add(ConstantsKeys.LATEST);
+           } else if (!StringUtils.isEmpty(locales)) {
+               for (String locale : locales.split(",")) {
+                   localeList.add(getMapingLocale(productName, version, locale.trim()));
+               }      
+           } else {
+               localeList = productService.getSupportedLocaleList(productName,version);
+           }
+           translationDTO.setLocales(localeList);
+           translationDTO.setPseudo(new Boolean(pseudo));
+           try {
+               translationDTO = multipleComponentsService
+                       .getMultiComponentsTranslation(translationDTO);
+               return  super.handleResponse(APIResponseStatus.OK, translationDTO);
+           }catch(Exception e) {
+               logger.info("do part of translation is available business");
+               translationDTO = multipleComponentsService.getV2MultiComponentsTranslation(translationDTO);
```

Comment:
why do this method again?

Reply:
this method is get the part translation and no translation feature